src/slashcommands/Play.ts

```diff
@@ -0,0 +1,133 @@
+import {
+	ApplicationCommandDataResolvable,
+	CommandInteraction,
+	CacheType,
+	MessageEmbed,
+	GuildMember,
+} from 'discord.js';
+import { ApplicationCommandOptionTypes } from 'discord.js/typings/enums';
+import { Bot } from '../Bot';
+import { SlashCommand } from './SlashCommand';
+import { player, playOptions } from '../index';
+import { QueryType } from 'discord-player';
+import { Option, Subcommand } from './Option';
+import { joinVoiceChannel } from '@discordjs/voice';
+
+export class Play implements SlashCommand {
+	name: string = 'play';
+	description = 'Play a song in the voice channel.';
+	options: (Option | Subcommand)[] = [
+		new Option(
+			'query',
+			'The song or playlist you want to queue',
+			ApplicationCommandOptionTypes.STRING,
+			true
+		),
+	];
+	registerData: ApplicationCommandDataResolvable = {
+		name: this.name,
+		description: this.description,
+	};
+	requiredPermissions: bigint[] = [];
+	async run(
+		bot: Bot,
+		interaction: CommandInteraction<CacheType>
+	): Promise<void> {
+		try {
+			const embed = new MessageEmbed().setColor('BLUE');
+
+			let member = interaction.member as GuildMember;
+			let state = member.voice.channel;
+
+			//if user is not connected
```

Comment:
You're performing these checks a lot, if we're sticking to DRY coding principles, move these checks into a resource file you can call whenever you need to